### PR TITLE
Migrate the Chip component to typescript

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Migrate the Chip component to typescript.

## Fixes

Fixes: #212.

### CODE REVIEW

File: src/components/Chip/Chip.tsx [hunk 1]
```diff
@@ -1,19 +1,32 @@
 import React from "react";
 import PropTypes from "prop-types";
 import { highlightSubString } from "../../utils";
+import type { KeyboardEvent, MouseEvent } from "react";
+
+export type Props = {
+  lead?: string;
+  onClick: (
```
Comment (Caleb-Ellis) [inline src/components/Chip/Chip.tsx:8]: I think this might be optional

**Review by Caleb-Ellis — APPROVED** [review]
LGTM, at least as far as this PR is concerned. But the `onClick` type is a bit weird. Where does `cloud` come from?

File: src/components/Chip/Chip.tsx [hunk 2]
```diff
@@ -1,19 +1,32 @@
...
+    event: MouseEvent<HTMLDivElement> | { lead: string; cloud: string }
```
Comment (bartaz) [inline src/components/Chip/Chip.tsx:9]: This event type seems weird. The lead/cloud thing seems project specific (I guess it's just a leftover from JAAS). And from implementation it seems that onClick works differently when clicking (returns event) and using keyboard (returns arbitrary object).

Probably out of scope for this migration PR, but we should have a separate issue to address this.

Comment (huwshimi) [reply src/components/Chip/Chip.tsx:9]: I've filed an issue https://github.com/canonical-web-and-design/react-components/issues/507.

File: src/components/Chip/Chip.tsx (hunk 1 above)
Comment (huwshimi) [reply src/components/Chip/Chip.tsx]: Yeah I think you're right, I've updated it.

**Comment by bartaz** [comment]
I think tests (snapshot) needs to be updated.

**Pull request merged**